util/git/client.go

```diff
@@ -385,20 +396,87 @@ func (m *nativeGitClient) LsRefs() (*Refs, error) {
 // runs with in-memory storage and is safe to run concurrently, or to be run without a git
 // repository locally cloned.
 func (m *nativeGitClient) LsRemote(revision string) (res string, err error) {
+	set := make(map[subModule]bool)
+	var submoduleRefs []string
+	if os.Getenv(common.EnvGitSubmoduleUpdateChanges) == "true" {
+		gitmodules, err := m.loadGitmodulesFile()
+		if err == nil {
+			scanner := bufio.NewScanner(gitmodules)
+			var subModules []subModule
+			for scanner.Scan() {
```

Comment:
Is there a reason that the logic for reading the `.gitmodules` is not put into `loadGitmodulesFile()`? I think it would make sense to encapsulate it there.